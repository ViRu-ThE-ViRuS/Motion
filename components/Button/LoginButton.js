import React, { Component } from 'react';

import {
    View,
    Text,
    StyleSheet,
    TouchableOpacity
} from 'react-native';

class LoginButton extends Component {
    constructor(props, bgc = "#fff") {
        super(props);
        this.type = "Login";
        this.styles = StyleSheet.create({
            container: {
                height: 50,
                alignSelf: "stretch",
                alignItems: "center",
                justifyContent: "center",
                backgroundColor: bgc,
                marginBottom: 10,
                marginLeft: 10,
                marginRight: 10
            },
            text: {
                fontFamily: "SourceSansPro_Regular",
                color: "#fff",
                fontSize: 15
            }
        })
    }


    render() {
        return (
            <TouchableOpacity style={this.styles.container} onPress={this.props.onPress}>
                <View>
                    <Text style={this.styles.text}>{this.type}</Text>
                </View>
            </TouchableOpacity>
        )
    }
}

export class FacebookLoginButton extends LoginButton {
    constructor(props) {
        super(props, "#3b5998");
        this.type = "Facebook";
    }
}

export class GoogleLoginButton extends LoginButton {
    constructor(props) {
        super(props, "#d34836");
        this.type = "Google";
    }
}

export class EmailLoginButton extends LoginButton {
    constructor(props) {
        super(props, "#5C7D87");
        this.type = "Email";
    }
}
