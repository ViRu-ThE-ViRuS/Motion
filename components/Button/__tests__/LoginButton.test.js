import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';

import { FacebookLoginButton, GoogleLoginButton, EmailLoginButton } from './../LoginButton';

describe('LoginButton tests', () => {
  it('FacebookLoginButton renders correctly', () => {
    const tree = renderer.create(
      <FacebookLoginButton />
    );

    expect(tree).toMatchSnapshot();
  });

  it('GoogleLoginButton renders correctly', () => {
    const tree = renderer.create(
      <GoogleLoginButton />
    );

    expect(tree).toMatchSnapshot();
  });

  it('EmailLoginButton renders correctly', () => {
    const tree = renderer.create(
      <EmailLoginButton />
    );

    expect(tree).toMatchSnapshot();
  });
});
