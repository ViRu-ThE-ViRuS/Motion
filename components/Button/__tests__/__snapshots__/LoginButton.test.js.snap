// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginButton tests EmailLoginButton renders correctly 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  collapsable={undefined}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "alignSelf": "stretch",
      "backgroundColor": "#5C7D87",
      "height": 50,
      "justifyContent": "center",
      "marginBottom": 10,
      "marginLeft": 10,
      "marginRight": 10,
      "opacity": 1,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <View>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#fff",
          "fontFamily": "SourceSansPro_Regular",
          "fontSize": 15,
        }
      }
    >
      Email
    </Text>
  </View>
</View>
`;

exports[`LoginButton tests FacebookLoginButton renders correctly 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  collapsable={undefined}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "alignSelf": "stretch",
      "backgroundColor": "#3b5998",
      "height": 50,
      "justifyContent": "center",
      "marginBottom": 10,
      "marginLeft": 10,
      "marginRight": 10,
      "opacity": 1,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <View>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#fff",
          "fontFamily": "SourceSansPro_Regular",
          "fontSize": 15,
        }
      }
    >
      Facebook
    </Text>
  </View>
</View>
`;

exports[`LoginButton tests GoogleLoginButton renders correctly 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  collapsable={undefined}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "alignSelf": "stretch",
      "backgroundColor": "#d34836",
      "height": 50,
      "justifyContent": "center",
      "marginBottom": 10,
      "marginLeft": 10,
      "marginRight": 10,
      "opacity": 1,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <View>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#fff",
          "fontFamily": "SourceSansPro_Regular",
          "fontSize": 15,
        }
      }
    >
      Google
    </Text>
  </View>
</View>
`;
